src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs

```diff
@@ -134,9 +136,15 @@ private Task ProjectPropertyChangedAsync(Tuple<ImmutableList<IProjectValueVersio
             if (projectRestoreInfo != null)
             {
                 _projectVsServices.Project.Services.ProjectAsynchronousTasks
-                    .RegisterCriticalAsyncTask(JoinableFactory.RunAsync(() => _solutionRestoreService
-                            .NominateProjectAsync(_projectVsServices.Project.FullPath, projectRestoreInfo, CancellationToken.None)),
-                            registerFaultHandler: true);
+                    .RegisterCriticalAsyncTask(JoinableFactory.RunAsync(async () =>
+                    {
+                        await _solutionRestoreService
+                               .NominateProjectAsync(_projectVsServices.Project.FullPath, projectRestoreInfo, CancellationToken.None)
+                               .ConfigureAwait(true);
```

Comment:
NominateProjectAsync returning a Task that restore is done is the late change. ConfigureAwait probably does not need to be set to true. I was unsure how the `RegisterCriticalAsyncTask` would interact with this -- that is the main change needed to prevent deadlocks.